libs/sdk-bindings/src/breez_sdk.udl

```diff
@@ -133,7 +133,8 @@ dictionary RouteHint {
 };
 
 dictionary LNInvoice {
-    string bolt11;
+    string? bolt11;
+    string? bolt12;
```

Comment:
I was also conflicted about the decision. The merging into `LNInvoice` made sense to me given that the invoices are functionally equivalent at the moment, given that we do not handle recurrence and blinded paths yet, plus the fact that CLN's [create_invoice](https://docs.corelightning.org/reference/lightning-createinvoice) command (which is not directly called, as it is more low-level) does exactly the same.

Despite that, some higher level commands like `lightning-invoice` do not allow us to create bolt12 invoices directly, therefore it would make sense to distinguish between those which can and can't with two different interfaces, then using enums to specify either one of them. 

Lastly, regarding the codebase, I agree that having to propagate the changes everywhere is not ideal, yet I've checked that altering `parse_invoice` would require us to make many changes (if not more) as well. At the end of the day though, I think it is worth going forward as you suggested.

Reply:
I agree and strongly feel that parsing bolt11 should return a different type than LNInvoice simply because it has different data (even though currently we don't expose/use it).
As for exposing parse_invoice in the udl I agree it is not really necessary (perhaps there are times when the user "knows" that she have a bolt11 in hand and can use this function directly rather than going through the input parser). Doesn't mean we have to do the same mistake twice if we are not sure.

How about:
1. Leave the parse_invoice as is for now (bolt11 only).
2. Not expose parsing bolt12 directly but only via the input parser
2. Use different structure for bolt12 invoices.

Reply:
I think 
- `LNInvoice` could be renamed to `Bolt11Invoice` to take away any ambiguity.
- `parse_invoice` could be renamed to `parse_bolt11_invoice` to make it clear what it's for. Maybe add a `parse_bolt12_invoice` function (internally) if necessary.
- Add a `Bolt12Invoice` type
- The input parser could probably use another branch to parse bolt12 invoices as well.